### ISSUE TITLE
wake up action doesn't work

### ISSUE THREAD

**Issue by tradexsrl** [Ylianst/MeshCentral#1078]
hi have done this test:
created a group defined as managed with software agent , installed the agent.shutdown the pc.
pc is setup for wake on lan. if i use a wake on lan software that send magic packet to that MACADDRESS the pc power up.
if in MC (with pc powered off) i select wakeup action it doesn't do nothing...
i have tried to create another group managed via amt no agent , add the same user , then i add that pc to that group (inserting all parameters for amt) but the device was not added...how can i solve these problems?

**Comment by Ylianst** [comment]
For wake-on-LAN, is your MeshCentral server in LAN, Hybrid or WAN mode? If in WAN mode, you need another device with the mesh agent connected to the server on the same subnet at the machine you are trying to wake up. This is because MeshCentral will tell your connected agents to broadcast the magic packet.

**Comment by tradexsrl** [comment]
is in hybrid mode to manage lan and internet pc
how to setup meshcentral to send magic packet ?
but i suspect if you look at another issue i open ,wol+amt doesn't work so meshcentral will be not able to wol the pc

**Comment by tradexsrl** [comment]
before enabling amt i have a server with a software that scripted send magic packet to the pc MACADDRESS and in this way many pc in the morning automatically powerup
now with amt enable is not reliable anymore because some pc does power up other never and pc that power up sometimes doesn't powerup
i can disable amt but i will loose al the support feature...

**Comment by Ylianst** [comment]
No need to disable Intel AMT. The type of WOL packet the MeshCentral uses will work just fine with Intel AMT enabled. If you create a WOL packet that is ICMP, then Intel AMT will get it and not wake the platform, but MeshCentral uses a UDP WOL packet.

What mode is your MeshCentral server in now?

**Comment by tradexsrl** [comment]
i have tested and with amt enable my software is no more able to powerup the pc via magic packet.
if amt disable everythings is back as normal.
my server is in hybrid mode , please read the thread in the other ticket that say with ping enable in amt , wol doesn't work but for me doesn't work even with that disabled 
i don't understand how MC send wol packet if there is not setup macaddress for each pc and how can i script it to powerup many pc at once

**Comment by jsastriawan** [comment]
You could use meshcmd to send amtpower command to each individual PC address. Have a look at meshcmd --help amtpower.

I have put serverside WSMAN stack but I haven't worked on it for a long time.

**Comment by Ylianst** [comment]
If you use a tool that sends ICMP WOL packets, then it will not work with Intel AMT enabled. When you setup the MeshAgent, MeshCentral will get the MAC address of all the devices. You can see it in the "Details" page.

MeshCMD can do Intel AMT power on, [it's available here](https://www.meshcommander.com/meshcommander/meshcmd).

**Comment by tradexsrl** [comment]
ah ok so it's confirmed that wol paket doesn't work anymore with amt enable....unfortunately the MAC address shown on the MC is not the real one because i use openxt as main OS for lenovo pc and then on top of that i have one windows vm 7 pro that has it's let me say virtual MAC address:
![immagine](https://user-images.githubusercontent.com/62424179/77788739-120a6b00-7062-11ea-8fee-0b6cafee2753.png)
i will look at meshcmd to find a way to batch script and schedule the whole office pc to power up like i do with old wol packet

**Comment by Ylianst** [comment]
So, you have not confirmed anything unless you indicate what type of WOL packet was used. Can you indicate that software you used to send the WOL packet or what type of WOL was used? Thanks.

**Comment by tradexsrl** [comment]
is not explicity written in the software i used for wol that is this: https://wol.aquilatech.com/
but in the github there are two thread where they are talking about icmp that if these are blocked on the firewall , software is not able to powerup the pc , so in my understanding this is a icmp wol packet

**Comment by tradexsrl** [comment]
i have played with meshcmd and with amtpower i was able to power on an amt enabled pc.
so i will add a batch file to schedule power on office pc's with amt enabled
many thank's

**Comment by tradexsrl** [comment]
i had to disable intel amt because meshcmd with power on command is not reliable and i'm not able to power on 90% of our pc's
meshcmd will check on dhcp or dns to find the pc? i have checked and dhcp ip record are the same as in dns...so i don't understand why it does not power up...
